source/modulo_core/include/modulo_core/concepts.hpp

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <geometry_msgs/msg/accel_stamped.hpp>
+#include <geometry_msgs/msg/pose_stamped.hpp>
+#include <geometry_msgs/msg/transform_stamped.hpp>
+#include <geometry_msgs/msg/twist_stamped.hpp>
+#include <geometry_msgs/msg/wrench_stamped.hpp>
+#include <sensor_msgs/msg/joint_state.hpp>
+#include <std_msgs/msg/bool.hpp>
+#include <std_msgs/msg/float64.hpp>
+#include <std_msgs/msg/float64_multi_array.hpp>
+#include <std_msgs/msg/int32.hpp>
+#include <std_msgs/msg/string.hpp>
+#include <tf2_msgs/msg/tf_message.hpp>
+
+#include "modulo_core/EncodedState.hpp"
+
+#include <state_representation/State.hpp>
+#include <state_representation/space/cartesian/CartesianState.hpp>
+#include <state_representation/space/joint/JointState.hpp>
+
+namespace modulo_core::concepts {
```

Comment:
@domire8 @eeberhard We should agree on concept names/naming convention before merging as well.

Reply:
This is not really exposed to the user and derived components right? Just to know the importance of naming

Reply:
Well, it is included in `Component.hpp` so it is exposed and can be used in that respect, yes. But it is not something the user needs to interact with to use the library (they will not have to provide them anywhere).

Reply:
I always enjoy thinking about names!

I think `CorePrimitiveT` could just be `PrimitiveT`.

I might also suggest defining a templated `concept StateT = std::derived_from<T, state_representation::State>;`

apart from that I think `CoreT` is pretty good (as then the union of `StateT<T> || PrimitiveT<T>`), the only others I could consider would be `BuiltInT` or `ModuloT` but they are not as good.

Finally I might prefer something like `GenericT` or `ExternalT` over `CustomT`, but it's not a super strong preference.